custom/icds_reports/management/commands/custom_ls_data_pull.py

```diff
@@ -0,0 +1,108 @@
+import csv
+
+from corehq.apps.es.forms import FormES
+from datetime import datetime
+from corehq.apps.users.models import CommCareUser
+from django.core.management.base import BaseCommand
+from django.db import connections
+
+from corehq.sql_db.connections import get_icds_ucr_citus_db_alias
+
+
+def _run_custom_sql_script(command, day=None):
+    db_alias = get_icds_ucr_citus_db_alias()
+    if not db_alias:
+        return
+
+    cursor = connections[db_alias].cursor()
+    cursor.execute(command, [day])
+
+    return cursor.fetchall()
+
+
+def state_details():
+    query = """
+        SELECT
+            awc.state_name as state_name,
+            awc.supervisor_name as supervisor_name,
+            l.supervisor_id as supervisor_id,
+            awc.supervisor_site_code as supervisor_site_code
+            FROM "agg_ls" l
+            LEFT JOIN "awc_location_local" awc
+                ON (
+                awc.supervisor_id = l.supervisor_id
+                AND l.aggregation_level=awc.aggregation_level )
+            WHERE l.month='2020-03-01' AND l.aggregation_level=4 AND l.num_supervisor_launched>0
+        """
+    return _run_custom_sql_script(query)
+
+
+class Command(BaseCommand):
+    help = "Run Custom Data Pull"
+
+    def get_users(self, users):
+        users = list(set(users))
+        user_location_details = {}
+        for user in users:
+            t = CommCareUser.get_by_user_id(user)
+            user_location_details.update({
+                user: t.location_id
+            })
+        print("================================User details fetched=========\n")
+        print(len(user_location_details))
+        print("===============================================================")
+        return user_location_details
+
+    def handle(self, **options):
+
+        query = FormES().domain('icds-cas').xmlns('http://openrosa.org/formdesigner/327e11f3c04dfc0a7fea9ee57d7bb7be83475309').submitted(gte=datetime(2020,3,1), lt=datetime(2020,4,1))
+        forms_list = query.run().hits
+
+        users = []
+        for form in forms_list:
+            users.append(form['form']['case']['@user_id'])
+        user_location_details = self.get_users(users)
+        location_details = state_details()
+        headers = ['state_name', 'supervisor_name', 'supervisor_site_code', 'visit_type', 'beneficiary_type', 'visit_count']
+        data_rows = [headers]
+        home_visit = ["due_for_delivery", "just_delivered", "six_to_eight_months","underweight_child", "recent_registered_preg", "ben_death"]
+        vhnd = ["pregnant_woman", "received_dpt1", "received_dpt3", "received_measles"]
+        fast_data_rows = {}
+        for location in location_details:
```

Comment:
As mentioned before, I think this wants to be a national aggregation so 10 rows total